gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -279,137 +276,138 @@ public GoogleCloudStorageImpl(GoogleCloudStorageOptions options, Credentials cre
+    }
+
+    this.storageOptions = options;
+    this.downscopedAccessTokenFn = downscopedAccessTokenFn;
+
+    this.httpStatistics = new StatisticsTrackingHttpRequestInitializer();
+
+    this.httpRequestInitializer =
+        options.isTraceLogEnabled()
+            ? new ChainingHttpRequestInitializer(
+                httpStatistics, httpRequestInitializer, new EventLoggingHttpRequestInitializer())
+            : new ChainingHttpRequestInitializer(httpStatistics, httpRequestInitializer);
 
+    if (httpTransport == null) {
+      httpTransport =
+          HttpTransportFactory.createHttpTransport(
+              options.getProxyAddress(),
+              options.getProxyUsername(),
+              options.getProxyPassword(),
+              Duration.ofMillis(options.getHttpRequestReadTimeout()));
+    }
     this.storage =
-        checkNotNull(
-            createStorage(options, this.httpRequestInitializer), "storage must not be null");
+        new Storage.Builder(
+                httpTransport, GsonFactory.getDefaultInstance(), this.httpRequestInitializer)
+            .setRootUrl(options.getStorageRootUrl())
+            .setServicePath(options.getStorageServicePath())
+            .setApplicationName(options.getAppName())
+            .build();
+
     this.storageRequestFactory = new StorageRequestFactory(storage);
 
     this.metricsRecorder =
-        MetricsSink.CLOUD_MONITORING == this.storageOptions.getMetricsSink()
+        MetricsSink.CLOUD_MONITORING == options.getMetricsSink()
             ? CloudMonitoringMetricsRecorder.create(options.getProjectId(), credentials)
             : new NoOpMetricsRecorder();
 
     // Create the gRPC stub if necessary;
-    if (this.storageOptions.isGrpcEnabled()) {
+    if (options.isGrpcEnabled()) {
       this.watchdog =
-          Watchdog.create(Duration.ofMillis(storageOptions.getGrpcMessageTimeoutCheckInterval()));
+          Watchdog.create(Duration.ofMillis(options.getGrpcMessageTimeoutCheckInterval()));
       this.storageStubProvider =
-          StorageStubProvider.newInstance(
-              this.storageOptions, this.backgroundTasksThreadPool, credentials);
+          StorageStubProvider.newInstance(options, backgroundTasksThreadPool, credentials);
     }
-    this.downscopedAccessTokenFn = downscopedAccessTokenFn;
-  }
-
-  /**
-   * Constructs an instance of GoogleCloudStorageImpl.
-   *
-   * @param options {@link GoogleCloudStorageOptions} to use to initialize the object.
-   * @param storage {@link Storage} to use for I/O.
-   * @param downscopedAccessTokenFn Function that generates downscoped access token.
-   */
-  @VisibleForTesting
-  public GoogleCloudStorageImpl(
-      GoogleCloudStorageOptions options,
-      Storage storage,
-      Function<List<AccessBoundary>, String> downscopedAccessTokenFn) {
-    this(options, storage, downscopedAccessTokenFn, tryGetCredentialsFromStorage(storage));
-
-    warnIfTracingEnabled(storageOptions.isTraceLogEnabled());
```

Comment:
why are we removing this check?

Reply:
Because we do not have constructor where it's applicable. I.e. this PR removes constructor that allows to override trace logging flag by supplying `Storage` object w/o trace logging enabled.